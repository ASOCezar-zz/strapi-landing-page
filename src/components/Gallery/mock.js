export default {
  background: false,
  title: 'My Gallery',
  description: 'Description Grid',
  grid: [
    {
      altText: 'Title 1',
      image: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Title 2',
      image: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Title 3',
      image: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Title 4',
      image: 'https://source.unsplash.com/random/800x800?r=6',
    },
    {
      altText: 'Title 5',
      image: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Title 6',
      image: 'https://source.unsplash.com/random/800x800?r=5',
    },
  ],
};
