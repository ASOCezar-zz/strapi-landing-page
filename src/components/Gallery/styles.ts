import styled, { css, keyframes } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.font.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

const bordersAnimation = keyframes`
  0%{border-radius: 100%; transform: scale(0)};
  50%{border-radius: 50%; transform: scale(1.5)}
  100%{border-radius: 0; transform: scale(1)};
`;

const animateElement = ({ number, show }) =>
  show &&
  css`
    animation: ${bordersAnimation} 500ms ${number}s ease forwards;
  `;

export const GridElement = styled.div`
  ${({ number, show }: GridType) => css`
    overflow: hidden;
      ${animateElement({ number, show })}
    }
  `}
`;

const showItem = keyframes`
  0% {transform: scale(0)};
  50% {transform: scale(1.5)};
  100% {transform: scale(1)}
`;

type GridType = {
  number: number;
  show: boolean;
};

const animateGrid = ({ number, show }: GridType) =>
  show &&
  css`
    animation: ${showItem} 350ms ${number}s ease forwards;
  `;

export const Image = styled.img<GridType>`
  ${({ number, show }) => css`
    transform: scale(0);
    width: 100%;

    ${animateGrid({ number, show })}
  `}
`;
