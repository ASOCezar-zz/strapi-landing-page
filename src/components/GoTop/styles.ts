import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 5rem;
    height: 5rem;

    bottom: 3rem;
    right: 2rem;

    z-index: 5;
    opacity: 0.6;

    border-radius: 50%;
  `}
`;
