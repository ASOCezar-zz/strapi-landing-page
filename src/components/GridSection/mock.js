export default {
  background: false,
  title: 'My Grid',
  description: 'Description Grid',
  grid: [
    {
      title: 'Title 1',
      text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.',
    },
    {
      title: 'Title 2',
      text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    },
    {
      title: 'Title 3',
      text: 'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. ',
    },
  ],
};
