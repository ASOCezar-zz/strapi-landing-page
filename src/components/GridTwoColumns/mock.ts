import { GridTwoColumnsProps } from '.';

export default {
  title: 'GridTwoColumns',
  text: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Mollitia non pariatur, illum harum explicabo porro quisquam obcaecati exercitationem fuga rerum veniam dolore modi in itaque accusantium, vel quos, expedita recusandae.`,
  image: 'assets/images/javascript.svg',
} as GridTwoColumnsProps;
