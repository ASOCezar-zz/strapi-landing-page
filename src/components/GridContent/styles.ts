import styled, { css, keyframes } from 'styled-components';

import { Title } from '../Heading/styles';

export type ContainerType = {
  show: boolean;
};

const showTitle = keyframes`
  to {transform: translateY(0)};
`;

const presentContent = ({ show }: ContainerType) =>
  show &&
  css`
    ${Title} {
      animation: ${showTitle} 300ms 1s ease-in-out forwards;
    }
  `;

export const Container = styled.div<ContainerType>`
  ${({ show }) => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    overflow: hidden;
    transition: all 1s 1s ease-in-out;

    ${Title} {
      transform: translateY(-200%);
    }

    ${presentContent({ show })};
  `}
`;

const showText = keyframes`
  to {opacity: 1};
`;

const presentText = ({ show }: ContainerType) =>
  show &&
  css`
    animation: ${showText} 300ms 1500ms ease-in-out forwards;
  `;

export const Html = styled.div<ContainerType>`
  ${({ theme, show }) => css`
    margin-block: ${theme.font.spacings.huge};
    opacity: 0;

    ${presentText({ show })};
  `}
`;
