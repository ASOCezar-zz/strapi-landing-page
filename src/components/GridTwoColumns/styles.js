import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.font.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.font.spacings.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.font.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
  `}
`;
