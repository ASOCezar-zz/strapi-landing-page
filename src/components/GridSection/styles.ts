import styled, { css, keyframes } from 'styled-components';
import { ContainerType } from '../GridContent/styles';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

const revealText = keyframes`
  from {transform: translateX(-101%)}
  to {transform: translateX(101%)}
`;

const fade = keyframes`
  to{opacity: 1};
`;

const revealTitle = ({ show }: ContainerType) =>
  show &&
  css`
    ::after {
      animation: ${revealText} 500ms ease-out forwards;
    }
    > ${HeadingContainer} {
      animation: ${fade} 320ms 600ms linear forwards;
    }
  `;

const revealDescription = ({ show }: ContainerType) =>
  show &&
  css`
    animation: ${fade} 320ms 600ms linear forwards;
  `;

export const Container = styled.div<ContainerType>`
  ${({ theme, show }) => css`
    position: relative;

    .reveal-container {
      position: relative;
      overflow: hidden;
      width: fit-content;
      ${revealTitle({ show })}
      ::after {
        content: '';
        position: absolute;
        background-color: white;
        inset: 0;
        display: block;
      }

      ${HeadingContainer} {
        opacity: 0;
      }
    }

    > ${TextComponent} {
      opacity: 0;
      ${revealDescription({ show })}
    }

    & ${TextComponent} {
      margin-bottom: ${theme.font.spacings.medium};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

const showItem = keyframes`
  from {transform: scale(0)}
  to {transform: scale(1)};
`;

type GridType = {
  number: number;
  show: boolean;
};

const animateGrid = ({ number, show }: GridType) =>
  show &&
  css`
    animation: ${showItem} 350ms ${number}s linear forwards;
  `;

export const GridElement = styled.div<GridType>`
  ${({ theme, number, show }) => css`
    transform: scale(0);
    ${animateGrid({ number, show })}
    & ${HeadingContainer} {
      position: relative;
      left: -7rem;

      @media ${theme.media.lteMedium} {
        left: 3.1rem;
      }
    }

    & ${HeadingContainer}:before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.huge};
      top: -2rem;
      left: -5rem;
      transform: rotate(12deg);
    }
  `}
`;
