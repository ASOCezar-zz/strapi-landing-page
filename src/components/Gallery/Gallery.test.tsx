import { Gallery } from '.';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Gallery />', () => {
  it('Should render a Gallery component and match with snapshot', () => {
    const { container } = renderTheme(<Gallery {...mock} />);
    expect(container).toBeInTheDocument;
    expect(container.firstChild).toMatchSnapshot();
  });
  it('Should render a Gallery component and match with snapshot', () => {
    const { container } = renderTheme(<Gallery {...mock} background={undefined} />);
    expect(container).toBeInTheDocument;
    expect(container.firstChild).toMatchSnapshot();
  });
});
