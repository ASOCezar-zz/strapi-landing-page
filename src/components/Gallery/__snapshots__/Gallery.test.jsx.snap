// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery /> Should render a Gallery component and match with snapshot 1`] = `
.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 5.6rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c3 {
  color: #0A1128;
  font-size: 5.6rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c5 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div
  class="c0"
  id=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        My Gallery
      </h2>
      <div
        class="c4 c5"
      >
        Description Grid
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <img
            alt="Title 1"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=1"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 2"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=2"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 3"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=3"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 4"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=6"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 5"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=4"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 6"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=5"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Gallery /> Should render a Gallery component and match with snapshot 2`] = `
.c5 {
  font-size: 2.4rem;
}

.c2 > .c4 {
  margin-bottom: 5.6rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c8:hover {
  -webkit-transform: scale(1.2) rotate(10deg);
  -ms-transform: scale(1.2) rotate(10deg);
  transform: scale(1.2) rotate(10deg);
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c3 {
  color: #0A1128;
  font-size: 5.6rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c5 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 4rem;
  }
}

<div
  class="c0"
  id=""
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3"
      >
        My Gallery
      </h2>
      <div
        class="c4 c5"
      >
        Description Grid
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <img
            alt="Title 1"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=1"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 2"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=2"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 3"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=3"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 4"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=6"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 5"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=4"
          />
        </div>
        <div
          class="c7"
        >
          <img
            alt="Title 6"
            class="c8"
            src="https://source.unsplash.com/random/800x800?r=5"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
